buildchain/buildchain/salt_tree.py

```diff
@@ -221,6 +221,10 @@ def _get_parts(self) -> Iterator[str]:
         renderer=targets.Renderer.JSON,
     ),
 
+    Path('salt/metalk8s/addons/addons-service-config/configmap.sls'),
+    Path('salt/metalk8s/addons/addons-service-config/deployed.sls'),
+    Path('salt/metalk8s/addons/addons-service-config/init.sls'),
```

Comment:
You should move these files, we said that state that should only run on the salt-master need to be called `deployed`.
Here you can call `addons.addons-service-config` and `addons.addons-service-config.configmap` and both will deploy kubernetes objects.

So here you should have
```
salt/metalk8s/addons/addons-service-config/deployed/init.sls
salt/metalk8s/addons/addons-service-config/deployed/configmap.sls
```
And that it, then you will be able to do `state.sls metalk8s.addons.addons-service-config.deployed`

Also IMO the `addons` in `addons-service-config` is a bit redundant